packages/strapi-design-system/src/Button/Button.js

```diff
@@ -54,7 +54,9 @@ export const Button = ({ variant, leftIcon, rightIcon, disabled, children, size,
           {leftIcon}
         </Box>
       )}
-      <Text small={size === 'S'}>{children}</Text>
+      <Text small={size === 'S'} as="span">
```

Comment:
ℹ️ I've fixed this on the fly since button can't hold a paragraph and Text is a p by default